### ISSUE TITLE
Feature Request: Scheduled posts

### ISSUE THREAD

**Issue by bdougie** [netlify/netlify-cms#263]
# This would be a great addition 
 
Serverless has a nice way to do scheduled posts. This would be a great addition to the CMS, but would require some sort of AWS connection, which does not currently exist.

Starting this discussion in hopes that it will get added to an eventual feature roadmap.

https://serverless.com/blog/static-site-post-scheduler/

**Comment by lgallen** [comment]
Would like to see this too. Another feature that is relied on for those migrating from the WordPress community.

**Comment by tech4him1** [comment]
The biggest barrier I see to this is actually posting the scheduled post without having the CMS open. Since we run in-browser only, there is no actual server to run the post.

**Comment by erquhart** [comment]
Right, Netlify CMS has no way to handle this itself. The OP example would use a Lambda function running a cron job, just as the linked article describes.

**Comment by blargity** [comment]
@erquhart could we do something with Netlify functions now that that's a thing?

**Comment by gil--** [comment]
i feel like something along the lines of https://www.nczonline.net/blog/2018/03/scheduling-jekyll-posts-netlify/ would work, although probably not with Netlify functions since it needs CloudWatch for the cron.

**Comment by emveeoh** [comment]
Ideally, every asset (a post, a page, an image, etc) would have the ability to be scheduled. Atomic scheduling.

**Comment by araphiel** [comment]
Is anyone already working on this?

If not, I can contribute.

**Comment by erquhart** [comment]
That'd be great - what do you have in mind?

**Comment by leomotta121** [comment]
Waiting for some news about this

**Comment by matrixik** [comment]
Please, stay open.

**Comment by austincondiff** [comment]
This is a fantastic idea. What if we somehow used GitHub actions in the users repo to accomplish this? I am happy to add this into the v3 UI mocks to show you how it might look visually if this is something we are looking into doing.

**Comment by erezrokah** [comment]
> This is a fantastic idea. What if we somehow used GitHub actions in the users repo to accomplish this?

This is a very good thought. I think the CMS could provide a UI over different services CI pipelines (where available) and auto generate flows to support scheduling.

**Comment by papistan** [comment]
For those looking for solutions in the meantime, if you have CI with scheduled workflow/builds (CircleCI & Travis), one idea to set up without Zapier: 
- schedule a repeating CI build every AM (or every hour)
- have a "Scheduled Post Time" input field where user enters future date/time
- check date/time before running `createPage` or whatever page building function to build all posts (so post /pages only build when date/time is in past)

**Comment by jwkicklighter** [comment]
Just wanted to add [another example](https://www.jerriepelser.com/blog/schedule-posts-gatsby-netlify/) I came across where someone is using GitHub Actions co accomplish this task.

Based on what @papistan said, I was thinking about a very similar idea for GitLab. One could hypothetically schedule a pipeline for ever X interval which has a single Job that looks at some criteria (perhaps a yml file in each Merge Request branch) and merges any branches that are ready to be published.

**Comment by kmairye** [comment]
Is there any news on this one? Having a use case where scheduled blog posts would relieve a lot of pain points.

**Comment by TheReyzer** [comment]
2023. Any news guys? Really need this!
